packages/browser/core/src/core/auth/internal/StitchAuthImpl.ts

```diff
@@ -140,12 +265,108 @@ export default class StitchAuthImpl extends CoreStitchAuth<StitchUser>
       });
     }
   }
+
+  public setRedirectFragment(key: RedirectKeys | RedirectFragmentFields, value: any) {
+    this.storageRef.set(key, value);
+  }
+}
+
+function generateState(): string {
+  let state = "";
+  for (let i = 0; i < 64; ++i) {
+    state += alphaNumericCharacters.charAt(Math.floor(Math.random() * alphaNumericCharacters.length));
+  }
+
+  return state;
+}
+
+export function prepareRedirect(
+  auth: StitchAuthImpl,
+  credential: StitchRedirectCredential,
+  redirectType: RedirectTypes
+): { redirectUrl: string; state: string } {
+  auth.setRedirectFragment(RedirectKeys.RedirectProvider, credential.providerType);
+  let redirectUrl = credential.redirectUrl;
+  if (redirectUrl === undefined) {
+    redirectUrl = pageRootUrl();
+  }
+
+  const state = generateState();
+  auth.setRedirectFragment(RedirectFragmentFields.State, state);
+  auth.setRedirectFragment(RedirectKeys.Type, redirectType);
```

Comment:
It looks like this isn't ever used again

Reply:
👍